### PR TITLE
Update tech-implementation.md

### PR DESCRIPTION
**Pull request by Christiandike**
Fix typos and grammatical errors. Refactor sentences. Use appropriate technical terms.

### CODE REVIEW

**Comment by KaluBuikem** [comment]
Please could this be assigned to me? To contribute to this project.

**Comment by Ginohmk** [comment]
@GeorgianaElena I am an outreachy applicant, and I really want to contribute to the jupyterhub project but all 21 issues seems to be on review.. I am proficient with python, CSS , html, JavaScript , 
Please how can I contribute ?

File: docs/source/rbac/tech-implementation.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Technical Implementation
 
-Roles are stored in the database, where they are associated with users, services, etc., and can be added or modified as explained in {ref}`define-role-target` section. Users, services, groups, and tokens can gain, change, and lose roles. This is currently achieved via `jupyterhub_config.py` (see {ref}`define-role-target`) and will be made available via API in future. The latter will allow for changing a token's role, and thereby its permissions, without the need to issue a new token.
+[Roles](https://jupyterhub.readthedocs.io/en/stable/rbac/roles.html#roles) are stored in the database, where they are associated with users, services, groups and tokens. Roles can be added or modified as explained in the {ref}`define-role-target` section. Users, services, groups, and tokens can gain, change, and lose roles. This is currently achieved via `jupyterhub_config.py` (see {ref}`define-role-target`) and will be made available via API in the future. The latter will allow for changing a token's role, and thereby its permissions, without the need to issue a new token.
```
Comment (minrk) [inline docs/source/rbac/tech-implementation.md:3]: We should use _internal_ markdown links, not full URLs:

```suggestion
[Roles](roles) are stored in the database, where they are associated with users, services, and groups. Roles can be added or modified as explained in the {ref}`define-role-target` section. Users, services, groups, and tokens can gain, change, and lose roles. This is currently achieved via `jupyterhub_config.py` (see {ref}`define-role-target`) and will be made available via API in the future. The latter will allow for changing a user's role, and thereby its permissions, without the need to restart JupyterHub.
```

File: docs/source/rbac/tech-implementation.md [hunk 2]
```diff
@@ -43,26 +43,25 @@ Prior to 3.0, tokens stored _roles_,
...
 the Hub logs a warning in this case (see {ref}`Figure 2 <api-request-chart>`).
 
-Resolving a token's scope (yellow box in {ref}`Figure 1 <token-request-chart>`) corresponds to resolving all the token's owner roles (including the roles associated with their groups) and the token's own scopes into a set of scopes. The two sets are compared (Resolve the scopes box in orange in {ref}`Figure 1 <token-request-chart>`), taking into account the scope hierarchy.
-If the token's scopes are a subset of the token owner's scopes, the token is issued with the requested scopes; if not, JupyterHub will raise an error.
+Resolving a token's scope (yellow box in {ref}`Figure 1 <token-request-chart>`) corresponds to resolving all the roles of the token's owner (including the roles associated with their groups) and the token's own scopes into a set of scopes. The two sets are compared (Resolve the scopes box in orange in {ref}`Figure 1 <token-request-chart>`), taking into account the scope hierarchy.
+If the token's scopes are a subset of the token owner's scopes, the token is issued with the requested scopes; if not, JupyterHub will throw an error.
```
Comment (minrk) [inline docs/source/rbac/tech-implementation.md:64]: Errors are raised in Python, not thrown:

```suggestion
If the token's scopes are a subset of the token owner's scopes, the token is issued with the requested scopes; if not, JupyterHub will raise an error.
```

**Review by minrk — COMMENTED** [review]
Great! I made a couple of suggestions in-line.

**Comment by Christiandike** [comment]
Thank you @minrk. I have applied the suggestions

**Comment by Christiandike** [comment]
@GeorgianaElena please can you merge this PR? I have applied the suggestions made by @minrk and he seems okay with everything else. I would really appreciate 🙏. Thanks

**Comment by GeorgianaElena** [comment]
> @GeorgianaElena please can you merge this PR? I have applied the suggestions made by @minrk and he seems okay with everything else.

Yes, this looks good to me too! Thanks a lot for this work @Christiandike! Merging now 🚀

**Pull request merged**